WordPress/gutenberg
# Modal: Convert component to TypeScript

## What?
Refactor the `Modal` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored the component TypeScript
- Refactored tests to TypeScript
- Refactored storybook from knobs to controls

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test the modal in the storybook.
3. Test the various use of Modals within the block/site editors in Gutenberg still work
